libsql-server/src/auth/mod.rs

```diff
@@ -0,0 +1,18 @@
+use std::sync::Arc;
+
+pub mod authenticated;
+pub mod authorized;
+pub mod constants;
+pub mod errors;
+pub mod parsers;
+pub mod permission;
+pub mod user_auth_strategies;
+
+pub use authenticated::Authenticated;
+pub use authorized::Authorized;
+pub use errors::AuthError;
+pub use parsers::{parse_http_auth_header, parse_http_basic_auth_arg, parse_jwt_key};
+pub use permission::Permission;
+pub use user_auth_strategies::{Disabled, HttpBasic, Jwt, UserAuthContext, UserAuthStrategy};
+
+pub type Auth = Arc<dyn UserAuthStrategy>;
```

Comment:
I think we can do this in a follow up and the code change should be pretty small, but I would like this to be a struct since type aliases tend to leak the internal implementation detail. So it could be a struct like:

```
#[derive(Clone)]
pub struct Auth {
    inner: Arc<dyn UserAuthStrategy + Send + Sync>,
}

impl Auth {
    // implement functions that we need to be public + constructor via `impl UserAuthStrategy`
}
```

Reply:
I'd like to take care of this in this pull request. I'll take a stab at it right now :D

Reply:
This is taken care of here [ad18ef0](https://github.com/tursodatabase/libsql/pull/1026/commits/ad18ef0f56d653b68c0270a94a452ce23e5cf766)